### PR TITLE
fill in identifiers for events

### PR DESCRIPTION
**Pull request by ndurell**
# Description
In updating the sdk to support the new event api changes we appear to have dropped the ids from some events in certain cases. It also looks like customer properties on events were dropped as well. This is now fixed and should get applied properly. Coincident with this PR we started seeing flakiness in the CI build. I've added some extra timeouts to help with that. Ideally though we should investigate this more deeply to determine the cause.

# Check List

- [ ] Are you changing anything with the public API?
- [x] Have you tested this change on real device?
- [x] Are your changes backwards compatible with previous SDK Versions?
- [x] Have you added unit test coverage for your changes?
- [x] Have you verified that your changes are compatible with all the operating system version this SDK currently supports?

# Manual Test Plan

1.

# Supporting Materials

### CODE REVIEW

**Review by jordan-griffin — APPROVED** [review]

File: Sources/KlaviyoSwift/Klaviyo.swift [hunk 1]
```diff
@@ -189,8 +189,7 @@ public class Klaviyo: NSObject {
            let body = properties["body"] as? [String: Any], let _ = body["_k"] {
             Self.sdkInstance
                 .create(event: Event(name: .OpenedPush,
-                                     properties: properties,
-                                     profile: [:]))
```
Comment (ndurell) [inline Sources/KlaviyoSwift/Klaviyo.swift:193]: This was not needed, was already an optional parameter.

File: Sources/KlaviyoSwift/KlaviyoModels.swift [hunk 2]
```diff
@@ -54,7 +54,7 @@ public struct Event: Equatable {
 
     private let _properties: AnyCodable
     public var profile: [String: Any] {
-        _profile.value as! [String: Any]
+        _profile.value as? [String: Any] ?? [:]
```
Comment (ndurell) [inline Sources/KlaviyoSwift/KlaviyoModels.swift:57]: I started seeing a crash here so I made this optional.

File: Sources/KlaviyoSwift/KlaviyoModels.swift [hunk 3]
```diff
@@ -70,7 +70,16 @@ public struct Event: Equatable {
                 value: Double? = nil,
                 time: Date? = nil,
                 uniqueId: String? = nil) {
-        _profile = AnyCodable(profile ?? [:])
+        var profile = profile
```
Comment (ndurell) [inline Sources/KlaviyoSwift/KlaviyoModels.swift:73]: This is probably temporary. We move some old parameters from the profile dictionary to the identifers structure so it gets picked up in the api request later. If any of these are absent they were be overwritten by the state value or they will overwrite state.

File: Tests/KlaviyoSwiftTests/APIRequestErrorHandlingTests.swift [hunk 4]
```diff
@@ -9,6 +9,8 @@
 import Foundation
 import XCTest
 
+let TIMEOUT_NANOSECONDS: UInt64 = 10_000_000_000 // 10 seconds
```
Comment (ndurell) [inline Tests/KlaviyoSwiftTests/APIRequestErrorHandlingTests.swift:12]: This test become flakey so I added some timeout here to reduce flakiness.

File: Tests/KlaviyoSwiftTests/KlaviyoStateTests.swift [hunk 5]
```diff
@@ -45,7 +45,7 @@ final class KlaviyoStateTests: XCTestCase {
     let TEST_INVALID_PROPERTIES_EVENT = [
         "properties": [
             1: "propValue"
-        ],
+        ] as [AnyHashable: String],
```
Comment (ndurell) [inline Tests/KlaviyoSwiftTests/KlaviyoStateTests.swift:48]: Cleans up a warning.

File: Tests/KlaviyoSwiftTests/KlaviyoTestUtils.swift [hunk 6]
```diff
@@ -156,7 +157,7 @@ extension StateChangePublisher {
     static let test = { () -> StateChangePublisher in
         StateChangePublisher.debouncedPublisher = { publisher in
             publisher
-                .debounce(for: .seconds(0), scheduler: DispatchQueue.main)
+                .debounce(for: .seconds(0), scheduler: DispatchQueue.immediate)
```
Comment (ndurell) [inline Tests/KlaviyoSwiftTests/KlaviyoTestUtils.swift:160]: I noticed i had used the main scheduler here. Using immediate should help speed things up.

File: Tests/KlaviyoSwiftTests/StateManagementTests.swift [hunk 7]
```diff
@@ -79,7 +79,7 @@ class StateManagementTests: XCTestCase {
         stateChangeSubject.send(completion: .finished)
         lifecycleSubject.send(completion: .finished)
 
-        wait(for: [stateChangeIsSubscribed, lifecycleExpectation], timeout: 1.0)
+        await fulfillment(of: [stateChangeIsSubscribed, lifecycleExpectation])
```
Comment (ndurell) [inline Tests/KlaviyoSwiftTests/StateManagementTests.swift:82]: fixes a warning.

File: Tests/KlaviyoSwiftTests/StateManagementTests.swift [hunk 8]
```diff
@@ -382,9 +382,23 @@ class StateManagementTests: XCTestCase {
...
         _ = await store.send(.enqueueEvent(event)) {
-            $0.email = "foo"
+            $0.email = "foo@foo.com"
+            $0.phoneNumber = "666BLOB"
+            $0.externalId = "my_user_id"
+            try $0.enqueueRequest(request: .init(apiKey: XCTUnwrap($0.apiKey), endpoint: .createEvent(.init(data: .init(event: event, anonymousId: XCTUnwrap($0.anonymousId))))))
+        }
+    }
+
+    func testEnqueueEventWithIdentifiers() async throws {
```
Comment (ndurell) [inline Tests/KlaviyoSwiftTests/StateManagementTests.swift:394]: Added another equivalent test that uses newer api structures.

File: Tests/KlaviyoSwiftTests/__Snapshots__/EncodableTests/testEventPayload.1.json [hunk 9]
```diff
@@ -15,7 +15,11 @@
             "anonymous_id" : "anon-id",
             "email" : "blob@email.com",
             "properties" : {
-
+              "email" : "blob@email.com",
```
Comment (ndurell) [inline Tests/KlaviyoSwiftTests/__Snapshots__/EncodableTests/testEventPayload.1.json:18]: This is expected and should have been caught earlier when we updated the api.

File: Tests/KlaviyoSwiftTests/__Snapshots__/KlaviyoStateTests/testLoadWithRequestMigration.1.json [hunk 10]
```diff
@@ -26,7 +26,7 @@
                       "email" : "value",
                       "external_id" : "value",
                       "properties" : {
-
+                        "foo" : "bar"
```
Comment (ndurell) [inline Tests/KlaviyoSwiftTests/__Snapshots__/KlaviyoStateTests/testLoadWithRequestMigration.1.json:29]: Same here we would have dropped some properties in migrated requests.

**Pull request merged**